### PR TITLE
Script dsl lesson rename

### PR DESCRIPTION
**Pull request by uponthesun**
The net result should be that the script DSL uses "lesson" instead of "stage.

The commits are hopefully separated logically so you can review each part of the change in isolation.

I don't think there should be any deployment issues but it'd be great to have your take on that too.

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
This looks awesome! The curriculum team will definitely appreciate a heads up on this.

one minor request is to remove "WIP" from the PR description.

you asked about deployment issues. I think it's generally ok to be a bit more lax about deployment issues which only affect curriculum writers, since they generally have to close out of levelbuilder during the deploy and reload after, so there's less risk of server + client code getting out of sync. plus, giving them a warning that this is happening will help them know that they might need to reload.

File: dashboard/test/models/script_test.rb [hunk 1]
```diff
@@ -227,7 +227,7 @@ def populate_cache_and_disconnect_db
     scripts, _ = Script.setup([script_file_all_properties])
     script = scripts.first
 
-    assert_equal 21, script.properties.keys.length
+    assert_equal 20, script.properties.keys.length
```
Comment (dmcavoy) [inline dashboard/test/models/script_test.rb:230]: Should this be changing?

Comment (uponthesun) [reply dashboard/test/models/script_test.rb:230]: Yea, I removed `hideable_stages` from the script_dsl parse output in this change also, forgot to call that out.

Comment (dmcavoy) [reply dashboard/test/models/script_test.rb:230]: No worries - Was just scanning for anything that jumped out. All looks good then!

**Review by dmcavoy — APPROVED** [review]

**Pull request merged**